### PR TITLE
cassandra-cpp-driver: add version 2.16.2, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **cassandra-cpp-driver/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cassandra-cpp-driver/all/conandata.yml [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
-    - base_path: "source_subfolder"
-      patch_file: "patches/2.15.3/fix-rapidjson.patch"
+    - patch_file: "patches/2.15.3/fix-cmake.patch"
+      patch_description: "use cci package"
+      patch_type: "conan"
+    - patch_file: "patches/2.15.3/fix-rapidjson.patch"
+      patch_description: "fix include path for cci package"
+      patch_type: "conan"
+    - patch_file: "patches/2.15.3/fix-atomic.patch"
+      patch_description: "add static_cast for atomic enum"
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/cassandra-cpp-driver/all/conandata.yml:30]: Thanks your advise!
I will update this recipe.

File: recipes/cassandra-cpp-driver/all/patches/2.15.3/remove-attribute-for-msvc.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
+diff --git a/src/third_party/sparsehash/src/sparsehash/internal/hashtable-common.h b/src/third_party/sparsehash/src/sparsehash/internal/hashtable-common.h
+index bac2b88..e802b5c 100644
+--- a/src/third_party/sparsehash/src/sparsehash/internal/hashtable-common.h
++++ b/src/third_party/sparsehash/src/sparsehash/internal/hashtable-common.h
+@@ -50,8 +50,13 @@
+ _START_GOOGLE_NAMESPACE_
+
+ template <bool> struct SparsehashCompileAssert { };
++#if defined(_MSC_VER)
++#define SPARSEHASH_COMPILE_ASSERT(expr, msg) \
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/cassandra-cpp-driver/all/patches/2.15.3/remove-attribute-for-msvc.patch:10]: @uilianries 
Without this patch, following message is occured.

```
.conan2\p\b\cassa8ae3c63d48e8d\b\src\src\third_party\sparsehash\src\sparsehash/internal/hashtable-common.h(166,3): error C2065: 'unused': 定義されていない識別子です。
```

Sorry for Japanese error message.
It means "'unused' is an undefined identifier".

[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/cassandra-cpp-driver/all/patches/2.15.3/remove-attribute-for-msvc.patch:10]: Thank you!

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by uilianries** [comment]
We are checking this OpenSSL error, don't worry.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**